src/app/features/faq/faq.page.ts

```diff
@@ -0,0 +1,45 @@
+import { Component } from '@angular/core';
+import { NavController } from '@ionic/angular';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { BehaviorSubject, fromEvent } from 'rxjs';
+import { tap } from 'rxjs/operators';
+import { BUBBLE_IFRAME_URL } from '../../shared/dia-backend/secret';
+import { NetworkService } from '../../shared/network/network.service';
+
+@UntilDestroy()
+@Component({
+  selector: 'app-faq',
+  templateUrl: './faq.page.html',
+  styleUrls: ['./faq.page.scss'],
+})
+export class FaqPage {
+  readonly networkConnected$ = this.networkService.connected$;
+  readonly iframeUrl = `${BUBBLE_IFRAME_URL}/version-v230207-sam/faq`;
+  readonly iframeLoaded$ = new BehaviorSubject(false);
+
+  constructor(
+    private readonly networkService: NetworkService,
+    private readonly navController: NavController
+  ) {}
+
+  ionViewDidEnter() {
+    this.processIframeEvents();
+  }
+
+  processIframeEvents() {
+    fromEvent(window, 'message')
+      .pipe(
+        tap(event => {
+          const postMessageEvent = event as MessageEvent;
+          if (postMessageEvent.data === 'iframe-on-load') {
+            this.iframeLoaded$.next(true);
+          }
+          if (postMessageEvent.data === 'iframe-back-button-clicked') {
+            this.navController.pop();
+          }
+        }),
+        untilDestroyed(this)
+      )
+      .subscribe();
+  }
+}
```

Comment:
**by GPT**

The code looks good. It is already using enums for post message events, which is great. However, here are some improvement suggestions:

-   Consider using a guard clause instead of an if statement for the postMessageEvent check to make it more concise.
-   Use `pop()` instead of `back()` in `iframe-back-button-clicked` case to pop the page off the navigation stack entirely, rather than just going back one step.

Reply:
@tammyyang :
- opinionated, will be changed when we merge with enums version
- make sense will be covered after completing [[issue] back button will back to wrong page (should in the explore tab not my capture tab)](https://app.asana.com/0/1201016280880500/1203600983882519/f)